lib/fluent/plugin/filter_masking.rb

```diff
@@ -11,18 +11,37 @@ def strToHash(str)
         eval(str)
       end
 
+      def myDig(input, path)
+        curr = input
+        for segment in path do
+          if curr != nil && curr.respond_to?(:dig)
+            if curr[segment] == nil
+              curr = curr[segment.to_s]
+            else
+              curr = curr[segment]
+            end
+          else
+            return nil
+          end
+        end
+        curr
+      end
       # returns the masked record
       # error safe method - if any error occurs the original record is returned
       def maskRecord(record)
         maskedRecord = record
         excludedFields = []
-        @fieldsToExcludeJSONPathsArray.each do | field |
-          field_value = record.dig(*field)
-          if  field_value != nil
-            excludedFields = excludedFields + field_value.split(',')
+        begin
+          @fieldsToExcludeJSONPathsArray.each do | field |
+            field_value = myDig(record, field)
+            if  field_value != nil
+              excludedFields = excludedFields + field_value.split(',')
+            end
           end
+        rescue Exception => e
```

Comment:
in what case will `myDig` throw an error? if the record is not there it just returns `nil`

Reply:
If it got a non hash

Reply:
if it's a non hash value then `curr.respond_to?(:dig)` will return false and `nil` will be returned from myDig

Reply:
Correction, it's for cases that the value obtained is not a string, so it will mask all in this case instead of skipping the masking.